VASP: Parse the number of steps correctly from OUTCAR

pmrv opened the pull request:
For VASP jobs output/generic/steps is generated in either of two ways:
1. vasprun.xml can be parsed -> np.arange(len(output/generic/energy_tot))
2. vasprun.xml cannot be parsed -> np.linspace(0, #trigger)

where #trigger is the number of times the string

FREE ENERGIE OF THE ION-ELECTRON SYSTEM (eV)

occurs in the OUTCAR file.

The latter leads to fractional output and always 50 entries in that array.  Both is contradictory to the first way and seems absurd.  This commit changes this to align with the first way.  Also this fixes a bug where the intent of the previous code seems to have been to multiply the step indices by the value of NBLOCK.  This however is not done in the first way.

For some reason there is an explicit unit test (that will now fail) that checks the output of `output/generic/steps` is always an array of length 50 with fractional values.  I cannot begin to understand why.  I would remove, but I'd like some input first.

pmrv committed:
VASP: Parse the number of steps correctly from OUTCAR

For VASP jobs output/generic/steps is generated in either of two ways:
1. vasprun.xml can be parsed -> np.arange(len(output/generic/energy_tot))
2. vasprun.xml cannot be parsed -> np.linspace(0, #trigger)

where #trigger is the number of times the string

FREE ENERGIE OF THE ION-ELECTRON SYSTEM (eV)

occurs in the OUTCAR file.

The latter leads to fractional output and always 50 entries in that
array.  Both is contradictory to the first way and seems absurd.  This
commit changes this to align with the first way.  Also this fixes a bug
where the intent of the previous code seems to have been to multiply the
step indices by the value of NBLOCK.  This however is not done in the
first way.

pmrv commented on pyiron_atomistics/vasp/outcar.py:
```diff
@@ -664,18 +664,16 @@ def get_steps(filename="OUTCAR", lines=None):
+            if nblock_trigger in line:
+                line = _clean_line(line)
+                n_block = int(line.split(nblock_trigger)[-1])
+        return n_block * np.arange(0, len(trigger_indices))
```
```suggestion
        return np.arange(0, len(trigger_indices) * n_block, n_block)
```

pmrv committed:
Use proper arange syntax

ligerzero-ai commented:
Wow, okay, wtf? Great catch. LGTM

ligerzero-ai reviewed (approved):
LGTM - perm granted to remove the test. It doesn't make sense to me either.

pmrv committed:
Split NBLOCK line correctly

pmrv committed:
Use regex to match instead

pmrv committed:
Do not keep unused list around

pmrv committed:
Remove previous weird test case

The old test checked that the output of the get_steps parser is always
the linspace(0,1) regardless of the actual number of steps in the
OUTCAR.  This replaces the test by a naive parser.

pmrv committed:
Use array comparison

pmrv committed:
Fix get_time

Previous implementation never actually found POTIM, but would iterate
over the whole file anyway.

pmrv committed:
Fix get_time test; same issue as get_steps

pmrv committed:
fix typos
